.github/workflows/requirements.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: requirements
+
+on: [pull_request]
+
+jobs:
+  requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install covalent
+        run: python -m pip install .
+
+      - name: Install packages
+        run: python -m pip install pip-check-reqs
+
+      - name: Check extra core requirements
+        run: pip-extra-reqs -r werkzeug covalent covalent_dispatcher covalent_ui
+
+      - name: Check missing SDK requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
```

Comment:
the covalent module shouldn't need to be excluded if we're using relative imports properly (with the exception of the local/dask plugins)

Reply:
ok, should the `from covalent.x import y` stuff be changed to relative imports in core code?

Reply:
yes the reason is that we don't want to create circular imports, so in the SDK everything should be relative